pkg/lib/cockpit-connect-ssh.tsx

```diff
@@ -0,0 +1,643 @@
+            <p>{_("The resulting fingerprint is fine to share via public methods, including email.")}</p>
+            <p>{_("If the fingerprint matches, click 'Trust and add host'. Otherwise, do not connect and contact your administrator.")}</p>
+        </>;
+    } else {
+        const fingerprint_help = <Popover bodyContent={
+            _("The resulting fingerprint is fine to share via public methods, including email. If you are asking someone else to do the verification for you, they can send the results using any method.")}>
+            <OutlinedQuestionCircleIcon />
+        </Popover>;
+        body = <>
+            <p>{cockpit.format(_("You are connecting to $0 for the first time."), address.address)}</p>
+            <ExpandableSection toggleText={ _("Verify fingerprint") }
+                                isExpanded={verifyExpanded}
+                                onToggle={(_ev, value) => setVerifyExpanded(value) }>
+                <div>{_("Run this command over a trusted network or physically on the remote machine:")}</div>
+                <ClipboardCopy isReadOnly hoverTip={_("Copy")} clickTip={_("Copied")} className="hostkey-verify-help hostkey-verify-help-cmds pf-v5-u-font-family-monospace">ssh-keyscan -t {key_type} localhost | ssh-keygen -lf -</ClipboardCopy>
+                <div>{_("The fingerprint should match:")} {fingerprint_help}</div>
+                <ClipboardCopy isReadOnly hoverTip={_("Copy")} clickTip={_("Copied")} className="hostkey-verify-help hostkey-fingerprint pf-v5-u-font-family-monospace">{error['host-fingerprint']}</ClipboardCopy>
+            </ExpandableSection>
+        </>;
+    }
+
+    const onAddKey = async () => {
+        setInProgress(true);
+        debug("onAddKey", error);
+        try {
+            await cockpit.script(ssh_add_key_sh, [host_key, "known_hosts"], { err: "message" });
+            dialogResult.resolve();
+        } catch (ex) {
+            setDialogError((ex as cockpit.BasicError).toString());
+            setInProgress(false);
+        }
+    };
+
+    return (
+        <Modal id="ssh-unknown-host-dialog" isOpen
+                position="top" variant="medium"
+                onClose={() => dialogResult.reject("cancel")}
+                title={title}
+                footer={<>
+                    <Button variant="primary" onClick={onAddKey} isLoading={inProgress} isDisabled={inProgress}>
+                        { submitText }
+                    </Button>
+                    <Button variant="link" className="btn-cancel" onClick={() => dialogResult.reject("cancel")}>
+                        { _("Cancel") }
+                    </Button>
+                </>}
+        >
+            <Stack hasGutter>
+                { dialogError &&
+                    // @ts-expect-error: cockpit-components-inline-notification.jsx needs typing
+                    <ModalError dialogError={dialogError} />}
```

Comment:
This added line is not executed by any test.

Reply:
belt-and-suspenders, this isn't reachable.